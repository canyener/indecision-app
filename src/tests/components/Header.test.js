import React from 'react'
import { shallow } from 'enzyme'

import Header from '../../components/Header'

test('Should render Header correctly without subtitle', () => {
  const wrapper = shallow(<Header />)

  expect(wrapper).toMatchSnapshot()
})

test('Should render Header correctly with subtitle', () => {
  const wrapper = shallow(<Header subtitle="test subtitle" />)
  expect(wrapper).toMatchSnapshot()
})

test('Should set default title prop', () => {
  
  const wrapper = shallow(<Header />)
  const h1 = wrapper.find('h1').at(0)

  expect(h1.text()).toBe('Indecision')
})

test('Should set title prop correctly', () => {
  const wrapper = shallow(<Header title="test title" />)
  const h1 = wrapper.find('h1').at(0)

  expect(h1.text()).toBe('test title')
})

test('Should set subtitle prop correctly', () => {
  const wrapper = shallow(<Header subtitle="test subtitle" />)
  const h2 = wrapper.find('h2').at(0)

  expect(h2.text()).toBe('test subtitle')
})
