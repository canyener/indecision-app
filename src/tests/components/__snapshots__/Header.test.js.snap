// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly with subtitle 1`] = `
<div
  className="header"
>
  <div
    className="container"
  >
    <h1
      className="header__title"
    >
      Indecision
    </h1>
    <h2
      className="header__subtitle"
    >
      test subtitle
    </h2>
  </div>
</div>
`;

exports[`Should render Header correctly without subtitle 1`] = `
<div
  className="header"
>
  <div
    className="container"
  >
    <h1
      className="header__title"
    >
      Indecision
    </h1>
  </div>
</div>
`;
